cherab/core/math/function/float/function6d/tests/test_arg.py

```diff
@@ -0,0 +1,54 @@
+# cython: language_level=3
+
+# Copyright 2016-2025 Euratom
+# Copyright 2016-2025 United Kingdom Atomic Energy Authority
+# Copyright 2016-2025 Centro de Investigaciones Energéticas, Medioambientales y Tecnológicas
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+"""
+Unit tests for the Arg6D class.
+"""
+
+import unittest
+from cherab.core.math.function.float.function6d.arg import Arg6D
+
+# TODO: expand tests to cover the cython interface
+class TestArg6D(unittest.TestCase):
+
+    def test_arg(self):
+        testvals = [-1e10, -7, -0.001, 0.0, 0.00003, 10, 2.3e49]
+        for x in testvals:
```

Comment:
You can avoid this high level of nesting throughout the tests:

`for (x, y, z, u, w, v) in itertools.product(testvals, repeat=6):`